node/rpc/src/chain_api.rs

```diff
@@ -80,3 +94,77 @@ pub(crate) async fn chain_block_messages<DB: BlockStore + Send + Sync + 'static>
     };
     Ok(ret)
 }
+
+pub(crate) async fn chain_get_tipset_by_height<DB: BlockStore + Send + Sync + 'static>(
+    data: Data<State<DB>>,
+    Params(params): Params<(ChainEpoch, TipsetKeys)>,
+) -> Result<TipsetJson, JsonRpcError> {
+    let height = params.0;
+    let tsk = params.1;
```

Comment:
```suggestion
    let (height, tsk) = params;
```
This is a lot cleaner imo than the indexing of the tuple individually